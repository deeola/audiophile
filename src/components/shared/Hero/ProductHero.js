import React from "react";
import { BlackBackground, Title } from "./Herostyling";

function ProductHero({ title }) {
  return (
    <BlackBackground>
      <Title data-testid='title'>{title}</Title>
    </BlackBackground>
  );
}

export default ProductHero;
