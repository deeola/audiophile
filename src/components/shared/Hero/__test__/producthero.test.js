import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ProductHero from "../ProductHero";

afterEach(cleanup);

it("renders product hero header correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ProductHero></ProductHero>, div);
});

it("renders correct title of product hero of speaker", () => {
  const { getByTestId } = render(<ProductHero title={"speaker"}></ProductHero>);
  expect(getByTestId("title")).toHaveTextContent("speaker");
});

it("renders correct title of product hero of headphone", () => {
  const { getByTestId } = render(
    <ProductHero title={"headphone"}></ProductHero>
  );
  expect(getByTestId("title")).toHaveTextContent("headphone");
});

it("renders correct title of product hero of earphone", () => {
  render(<ProductHero title={"earphone"} />);
  const MyElemeng = screen.getByTestId("title");
  expect(MyElemeng).toHaveTextContent("earphone");
});

//Snapshots

it("matches product hero snapshot", () => {
  const tree = renderer
    .create(<ProductHero title={"Speaker"}></ProductHero>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
