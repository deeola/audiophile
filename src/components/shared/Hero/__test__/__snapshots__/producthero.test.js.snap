// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches product hero snapshot 1`] = `
<div
  className="sc-bczRLJ VOALK"
>
  <h1
    className="sc-dkzDqf jjKiLw"
    data-testid="title"
  >
    Speaker
  </h1>
</div>
`;
